packages/sui-cz/index.js

```diff
@@ -1,160 +1,13 @@
-              .then(result => {
-                if (typesWithOtherScopes.indexOf(answers.type) > -1) {
-                  return result.concat(otherScopes)
-                }
-                return result
-              })
-          }
-        },
-        {
-          type: 'input',
-          name: 'subject',
-          message:
-            'Write a SHORT, IMPERATIVE tense description of the change:\n',
-          validate: function (value) {
-            return !!value
-          },
-          filter: function (value) {
-            return value.charAt(0).toLowerCase() + value.slice(1)
-          }
-        },
-        {
-          type: 'input',
-          name: 'body',
-          message:
-            'Provide a LONGER description of the change (optional). Use "|" to break new line:\n'
-        },
-        {
-          type: 'input',
-          name: 'breaking',
-          message: 'List any BREAKING CHANGES (optional):\n',
-          when: function (answers) {
-            return allowBreakingChanges.indexOf(answers.type.toLowerCase()) > -1
-          }
-        },
-        {
-          type: 'input',
-          name: 'footer',
-          message:
-            'List any ISSUES CLOSED by this change (optional). E.g.: #31, #34:\n'
-        },
-        {
-          type: 'expand',
-          name: 'confirmCommit',
-          choices: [
-            { key: 'y', name: 'Yes', value: 'yes' },
-            { key: 'n', name: 'Abort commit', value: 'no' },
-            { key: 'e', name: 'Edit message', value: 'edit' }
-          ],
-          message: function (answers) {
-            var SEP =
-              '###--------------------------------------------------------###'
-            console.log(
-              '\n' + SEP + '\n' + buildCommit(answers) + '\n' + SEP + '\n'
-            )
-            return 'Are you sure you want to proceed with the commit above?'
-          }
-        }
-      ])
-      .then(function (answers) {
-        if (answers.confirmCommit === 'edit') {
-          temp.open(null, function (err, info) {
-            if (err) {
-              return
-            }
-
-            fs.write(info.fd, buildCommit(answers))
-            fs.close(info.fd, function () {
-              editor(info.path, function (code, sig) {
-                if (code === 0) {
-                  var commitStr = fs.readFileSync(info.path, {
-                    encoding: 'utf8'
-                  })
-                  commit(commitStr)
-                } else {
-                  console.log(
-                    'Editor returned non zero value. Commit message was:\n' +
-                      buildCommit(answers)
-                  )
-                }
-              })
-            })
-          })
-        } else if (answers.confirmCommit === 'yes') {
-          commit(buildCommit(answers))
-        } else {
-          console.log('Commit has been canceled.')
-        }
-      })
+    ErrorCommitSaver.retrievePreviousCommit(commitString => {
```

Comment:
Why not start always with the normal flow. and only perform the recover if you add a flag? like --recover or similar?

I normally cancel because I've made a mistake, this will make that flow slower every time.

Reply:
The flow doesn't be affected in performance terms if you are talking about that. 
On the other hand, this will only save the commit if you reached the last step and you've done a commit. Not in cancel, cmd+c or other cases.